dev/NavigationView/NavigationView.cpp

```diff
@@ -379,10 +427,354 @@ void NavigationView::OnApplyTemplate()
+            if (ir == m_topNavRepeater.get())
+            {
+                nvibImpl->Position(NavigationViewListPosition::TopPrimary);
+
+            }
+            else
+            {
+                nvibImpl->Position(NavigationViewListPosition::TopOverflow);
+            }
+        }
+        else
+        {
+            nvibImpl->Position(NavigationViewListPosition::LeftNav);
+        }
+
+        // Apply any custom container styling
+        ApplyCustomMenuItemContainerStyling(nvib, ir, args.Index());
+
+        if (auto nvi = args.Element().try_as<winrt::NavigationViewItem>())
+        {
+            auto nviImpl = winrt::get_self<NavigationViewItem>(nvi);
+            nviImpl->ClearIsContentChangeHandlingDelayedForTopNavFlag();
+            nviImpl->UseSystemFocusVisuals(ShouldShowFocusVisual());
+
+            // Register for item events
+            auto nviRevokers = winrt::make_self<NavigationViewItemRevokers>();
+            nviRevokers->tappedRevoker = nvi.Tapped(winrt::auto_revoke, { this, &NavigationView::OnNavigationViewItemTapped });
+            nviRevokers->keyDownRevoker = nvi.KeyDown(winrt::auto_revoke, { this, &NavigationView::OnNavigationViewItemKeyDown });
+            nviRevokers->keyUpRevoker = nvi.KeyUp(winrt::auto_revoke, { this, &NavigationView::OnNavigationViewItemKeyUp });
+            nviRevokers->gotFocusRevoker = nvi.GotFocus(winrt::auto_revoke, { this, &NavigationView::OnNavigationViewItemOnGotFocus });
+            nviRevokers->isSelectedRevoker = RegisterPropertyChanged(nvi, winrt::SelectorItem::IsSelectedProperty(), { this, &NavigationView::OnNavigationViewItemIsSelectedPropertyChanged });
+            nvi.SetValue(GetNavigationViewItemRevokersProperty(), nviRevokers.as<winrt::IInspectable>());
+        }
+    }
+}
+
+void NavigationView::ApplyCustomMenuItemContainerStyling(winrt::NavigationViewItemBase nvib, winrt::ItemsRepeater ir, int index)
+{
+    if (auto menuItemContainerStyle = MenuItemContainerStyle())
+    {
+        nvib.Style(menuItemContainerStyle);
+    }
+    else if (auto menuItemContainerStyleSelector = MenuItemContainerStyleSelector())
+    {
+        auto itemsSourceView = ir.ItemsSourceView();
+        auto item = itemsSourceView.GetAt(index);
+        auto selectedStyle = menuItemContainerStyleSelector.SelectStyle(item, nvib);
+        if (selectedStyle)
+        {
+            nvib.Style(selectedStyle);
+        }
+    }
+}
+
+void NavigationView::RepeaterElementClearing(winrt::ItemsRepeater ir, winrt::ItemsRepeaterElementClearingEventArgs args)
+{
+    if (auto nvi = args.Element().try_as<winrt::NavigationViewItem>())
+    {
+        auto nviImpl = winrt::get_self<NavigationViewItem>(nvi);
+        nviImpl->ClearIsContentChangeHandlingDelayedForTopNavFlag();
+        // Revoke all the events that we were listing to on the item
+        nvi.SetValue(GetNavigationViewItemRevokersProperty(), nullptr);
+    }
+
+    // We want to unlink the containers from the parent repeater
```

Comment:
>unlink [](start = 18, length = 6)

Is there no way to do this through the repeater APIs?

Reply:
I had a discussion about this with @ranjeshj sometime back and I don't believe there is a way to do this through the repeater APIs.
 The problem is that Repeater only unlinks containers if there is no ItemTemplate set. This is never the case for NavigationView because there is a custom element factory that is always set on the Repeaters so that data can be automatically wrapped in NVI if need be. 
This unlinking behavior is needed because repeater holding on to the NVI becomes a problem when the passed in items are NVI themselves, which prevents the control from moving them between repeaters.

Reply:
FYI, I ended up moving this unlinking logic to the `NavigationViewItemsFactory` recycle method